src/lib/eco-router/api.ts

```diff
@@ -151,6 +152,21 @@ export async function getExactIn(
       })
   })
 
+  // ZeroX
+  const zeroXTrade = new Promise<ZeroXTrade | undefined>(async resolve => {
+    if (!RoutablePlatform.ZEROX.supportsChain(chainId)) {
+      return resolve(undefined)
+    }
+
+    ZeroXTrade.bestTradeExactIn(currencyAmountIn, currencyOut, maximumSlippage)
+      .then(resolve)
+      .catch(error => {
+        errors.push(error)
+        resolve(undefined)
+        console.error(error)
+      })
```

Comment:
I'm seeing this code repeating too many times, and in some cases we don't push `error` or don't log it. It would be awesome to have a function to handle this error and we just right this code once.

Reply:
ok, I will refactor it in the separate PR.